### PR TITLE
Core(Moderation): Upgrade to v8.0.0

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: src/commands/Moderation/ban.ts [hunk 1]
```diff
@@ -23,10 +18,13 @@ export default class extends ModerationCommand {
...
-		await message.guild!.members.ban(target.id, { days: Number(message.flagArgs.day || message.flagArgs.days) || 0, reason });
-
-		return this.sendModlog(message, target, reason, null, duration);
+	public handle(message: KlasaMessage, target: User, reason: string | null, duration: number | null) {
+		return message.guild!.security.actions.ban({
+			user_id: target.id,
+			moderator_id: message.author.id,
+			duration,
+			reason
+		}, Number(message.flagArgs.day || message.flagArgs.days) || 0, this.getTargetDM(message, target));
```
Comment (vladfrangu) [inline src/commands/Moderation/ban.ts:27]: Just a tiny nitpick, this won't pickup `DAY` or `DAYS`

File: src/lib/structures/ModerationManagerEntry.ts [hunk 2]
```diff
@@ -87,30 +146,52 @@ export class ModerationManagerEntry {
 			: this.user.id;
 	}
 
+	public get appealTaskName() {
+		if (!this.appealable) return null;
+		switch (this.typeVariation) {
+			case Moderation.TypeVariation.Warning: return 'moderationEndWarning';
```
Comment (vladfrangu) [inline src/lib/structures/ModerationManagerEntry.ts]: Could turn these into const enums?

File: src/lib/structures/ModerationTask.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		} catch { /* noop */ }
+	}
+
+	protected async reschedule(data: ModerationData, duration: number) {
+		await this.client.schedule.create(this.name, Date.now() + duration, {
+			data: {
+				[Moderation.SchemaKeys.Guild]: data.guildID,
+				[Moderation.SchemaKeys.User]: data.userID,
+				[Moderation.SchemaKeys.Duration]: data.duration,
+				scheduleRetryCount: (data.scheduleRetryCount || 0) + 1
```
Comment (vladfrangu) [inline src/lib/structures/ModerationTask.ts:32]: You never check or use this

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: src/lib/structures/ModerationTask.ts (hunk 3 above)
Comment (kyranet) [reply src/lib/structures/ModerationTask.ts:32]: That is implemented for debug purposes.

File: src/commands/Moderation/ban.ts (hunk 1 above)
Comment (kyranet) [reply src/commands/Moderation/ban.ts:27]: What's your suggested solution?

File: src/lib/structures/ModerationManagerEntry.ts (hunk 2 above)
Comment (kyranet) [reply src/lib/structures/ModerationManagerEntry.ts]: Fixed in 8ddd565

File: src/commands/Moderation/ban.ts (hunk 1 above)
Comment (kyranet) [reply src/commands/Moderation/ban.ts:27]: Will be addressed on #460 with another PR.

File: src/languages/es-ES.ts [hunk 4]
```diff
@@ -2830,6 +2833,55 @@ export default class extends Language {
...
+			: `[Action] Applied Attachment Restriction | Reason: ${reason}`,
+		ACTION_RESTRICTED_VOICE_REASON: reason => reason === null
+			? '[Action] Applied Voice Restriction.'
+			: `[Action] Applied Voice Restriction | Reason: ${reason}`,
+		ACTION_REQUIRED_MEMBER: 'The user does not exist or is not in this server.',
+		MUTE_NOT_CONFIGURED: 'The muted role must be configured for this action to happen.',
+		MUTE_NOT_IN_MEMBER: 'The muted role is not set in the member.',
+		MUTE_LOW_HIERARCHY: 'I cannot mute a user which higher role hierarchy than me.',
+		MUTE_CANNOT_MANAGE_ROLES: `I must have **${PERMS.MANAGE_ROLES}** permissions to be able to mute.`,
+		MUTE_NOT_EXISTS: 'The specified user is not muted.',
```
Comment (favna) [inline src/languages/es-ES.ts:2883]: These should be translated, or an issue created to translate them later.

**Review by favna — COMMENTED** [review]
Found just one thing

**Pull request merged**